packages/interfaces/src/metadata.ts

```diff
@@ -1,8 +1,11 @@
 import type { PeerId } from "@libp2p/interface/peer-id";
 
 import type { ShardInfo } from "./enr.js";
-import type { IBaseProtocol } from "./protocols.js";
+import type { IBaseProtocol, ShardingParams } from "./protocols.js";
 
-export interface IMetadata extends IBaseProtocol {
+// IMetadata always has shardInfo defined while it is optionally undefined in IBaseProtocol
```

Comment:
nit: this comment is not needed as `type` is descriptive enough unless you add a reason for it